src/outputs/outputs.cpp

```diff
@@ -418,8 +418,9 @@ void Outputs::MakeOutputs(Mesh *pm, ParameterInput *pin, SimTime *tm) {
   bool first = true;
   OutputType *ptype = pfirst_type_;
   while (ptype != nullptr) {
-    if (tm == nullptr || (tm->time == tm->start_time) ||
-        (tm->time >= ptype->output_params.next_time) || (tm->time >= tm->tlim)) {
+    if ((ptype->output_params.dt >= 0.0) &&
```

Comment:
With this change, does the calculate_pi example still produce the hdf5 output it did before?

Reply:
Correct. I fixed the logic so that the test for `dt >=0` only occurs if the `tm` object exists.